src/shogun/regression/svr/LibLinearRegression.cpp

```diff
@@ -43,11 +45,19 @@ void CLibLinearRegression::init_defaults()
 
 void CLibLinearRegression::register_parameters()
 {
-	SG_ADD(&m_C, "m_C", "regularization constant",ParameterProperties::HYPER);
-	SG_ADD(&m_epsilon, "m_epsilon", "tolerance epsilon");
-	SG_ADD(&m_epsilon, "m_tube_epsilon", "svr tube epsilon",ParameterProperties::HYPER);
-	SG_ADD(&m_max_iter, "m_max_iter", "max number of iterations");
-	SG_ADD(&m_use_bias, "m_use_bias", "indicates whether bias should be used");
+	SG_ADD(&m_C, "C", "regularization constant", ParameterProperties::HYPER);
+	SG_ADD(
+		&m_epsilon, "m_tube_epsilon", "svr tube epsilon",
+		ParameterProperties::HYPER);
+	SG_ADD(&m_max_iter, "max_iter", "max number of iterations");
+	SG_ADD(&m_use_bias, "use_bias", "indicates whether bias should be used");
+	SG_ADD(
+		(machine_int_t*)&m_liblinear_regression_type,
+		"liblinear_regression_type", "Type of LibLinear regression.");
+	m_enum_to_string_map[get_name()]["liblinear_regression_type"] = {
```

Comment:
Actually, since this happens at runtime now, we could rather have one options map for each class, stored locally, and then a method “add_option(string, int)” that is called this method here. Or even better
“add_options(string param, map<string, int>)” which then can be called by directly writing out the map as {“foo”: SOLVER::FOO, “bar”: SOLVER::BAR} in a single call
I didn’t realise yesterday that the map doesn’t need to be gobal nor static

Reply:
Ah yes good point, I didn't quite like the fact that there was this static variable around! So `add_options(string param, map<string, int>)` just sits in SGObject, and then just need a mapping from param to string_enum_map, which makes it much simpler.

Reply:
Or maybe even: Do you think using a macro, we could avoid having to write the enum value name AND a string ?
Like
ADD_OPTION(param_name, FOO); which adds the mapping “FOO”->FOO where FOO is one option of the enum?

Reply:
This means we’d have to write things one by one but on the other hand it avoids the redundancy of explicitly naming the option string

Reply:
You should be able to pass the literal variable name in a macro yes. It's a shame there isn't a universal name demangling across compilers, would be quite useful in this case...

Reply:
yes, c++ lacks quite some things ... :scream: